WIP get timeout from R options

ReDressCodes opened the pull request:
This patch is yet to be tested.

Made the PR for any possible comments anyway.

ReDressCodes committed:
get timeout from R options

nfultz commented on R/client.R:
```diff
@@ -6,20 +6,21 @@
 grpc_client <- function(impl, channel) {
-  
+
+    timeout <- getOption("grpc.timeout")
```
You can add a default here (5000), otherwise existing code will need to be updated.

nfultz reviewed (commented):
Needs some documentation / testing, but code looks good.

ReDressCodes committed:
Add
 1) Default client timeout value which is necessary to avoid breaking already
  existing code which had already been hardcoded to five seconds.

2) Introduce a constant whose value is the number of milliseconds in one second

ReDressCodes committed:
Documentation for the previous changes

ReDressCodes committed:
Merge github.com:ghuoul/grpc

ghuoul commented:
Finally got around to testing this out.

Setting 'timeout' to zero, via the option 'grpc.timeout' means no response from the server. Calling a client twice does break the server, though.

It is admittedly a bit hard to document this, probably because the change is too minor but it looks to me like running devtools::document() and Rcpp::CompileAttributes() should update both documents and C++ exports to R.
